commands/Moderation/mute.js

```diff
@@ -0,0 +1,28 @@
+exports.run = async (client, msg, [member, role]) => {
+  role = typeof role === "string" ? msg.guild.roles.find("name", role) : role;
```

Comment:
This makes no sense to me, it's a mute command, not a giverole/removerole command. Use SettingGateway instead.

Reply:
What I was doing: to mute someone you have to create a role and assign/remove that specific role. This role is placed at the top of hierarchy and each channel has this role muted.

is there a better way to do this?

Reply:
The complete solution is extremely complex, you're required to store modlogs capable of storing "extraData" (in this case, array of role ids) so when you mute somebody, you take out all the roles, and when you do unmute, you retrieve the role ids and give all them back.

Just use SettingGateway to store the Muted role.

Reply:
@kyranet I could have sworn that's what I had before in mute command. But I was told this way is better and to replace it with this method